diff --git a/lib/env.js b/lib/env.js
index 1f95a3be..b6fac399 100644
--- a/lib/env.js
+++ b/lib/env.js
@@ -62,12 +62,11 @@ module.exports.create = function (config) {
 
     var httpSafeOrigin;
     var NO_SANDBOX = false;
-    var httpSafePort;
     var httpPort = isValidPort(config.httpPort)? config.httpPort: 3000;
+    var httpSafePort = isValidPort(config.httpSafePort) ? config.httpSafePort : (httpPort + 1);
 
     if (typeof(config.httpSafeOrigin) !== 'string') {
         NO_SANDBOX = true;
-        if (typeof(config.httpSafePort) !== 'number') { httpSafePort = httpPort + 1; }
         httpSafeOrigin = deriveSandboxOrigin(httpUnsafeOrigin, httpSafePort);
     } else {
         httpSafeOrigin = canonicalizeOrigin(config.httpSafeOrigin);
@@ -108,7 +107,7 @@ module.exports.create = function (config) {
         permittedEmbedders: typeof(permittedEmbedders) === 'string' && permittedEmbedders? permittedEmbedders: httpSafeOrigin,
 
         removeDonateButton: config.removeDonateButton,
-        httpPort: isValidPort(config.httpPort)? config.httpPort: 3000,
+        httpPort: httpPort,
         httpAddress: typeof(config.httpAddress) === 'string'? config.httpAddress: 'localhost',
         websocketPath: config.externalWebsocketURL,
